explainers/hidden-content-explainer.md

```diff
@@ -135,56 +128,45 @@ sections already has an event handler to expand and show the section. With the
-page receives an event.
-
-We believe that the benefit of providing this information to the site outweighs
-the potential risks. Moreover, the lower granularity information can already be
-approximated from scroll position and intersection observerations on the page.
-
-There are three cases to consider, each of which correspond to an action that
-caused the `beforematch` event to be fired:
-1. Find-in-page. The data typed into the search dialog for find-in-page is not
-   directly observable by the page. However, based on the `beforematch` event, the page
-   can infer that the searched text is within a particular element that received
-   the event. We believe that the risk of exposing this information to the page
-   is low:
-     * The granularity of information provided is low, since the event is
-       received by the containing element, and does not reveal the actual text
-       searched.
-     * Find-in-page shortcuts (such as CTRL-F) can already be intercepted by the
-       page, giving it opportunity to intercept user-typed "find-in-page"
-       requests and provide custom behavior.
-     * The position of the searched text can already be deduced by the page
-       based on scroll offsets and intersection observations.
-2. Scroll to text navigation. Depending on the implementation of scroll-to-text,
-   from the page's perspective the behavior is indistinguishable from
-   find-in-page behavior. This means that it reveals where the searched text is
-   located but not what it is. Furthemore, this information is only available to
-   the page using the `beforematch` event, and is not observable by any third
-   parties.
-
-Some consideration needs to be given to the `beforematch` event when it comes to
-cross frame interactions. Specifically, if the matched information is found within
-a frame, should the signal propagate to the embedding page?
-
-In general, we believe that the `beforematch` event does not pose a privacy problem.
-However, the privacy aspect of the event should be discussed in more detail via
-formal security reviews.
+The beforematch event could expose more information to the page than is
+currently exposed because it very directly hints to the text the user is
+searching for since it gets fired on the element containing the target text.
+
+Note that it is already possible to snoop on find-in-page by creating a very
+small scrollable area containing every next character the user could type into
+find-in-page, listening to scroll events to see which character the user typed
+in, then prepending the new character to all of the next possible search terms.
+This is demonstrated in
+[search-incremental.html](/resources/find-in-page/search-incremental.html).
+
+In order to mitigate the additional information that beforematch could expose to
+the page, we have added two constraints to when beforematch can be fired:
+
+1. The beforematch event will only be fired on `content-visibility:
```

Comment:
on content in the DOM subtree of an element with content-visibility: hidden-matchable.

(it's a bit unclear what content-visibility: hidden-matchable content is)

Reply:
done